docs/3.api/1.composables/use-async-data.md

```diff
@@ -51,7 +51,7 @@ type AsyncDataRequestStatus = 'idle' | 'pending' | 'success' | 'error'
 ## Params
 
 * **key**: a unique key to ensure that data fetching can be properly de-duplicated across requests. If you do not provide a key, then a key that is unique to the file name and line number of the instance of [`useAsyncData`](/docs/api/composables/use-async-data) will be generated for you.
-* **handler**: an asynchronous function that returns a value
+* **handler**: an asynchronous function that must return a value that should be different from `undefined` to avoid request duplication on the client side
```

Comment:
```suggestion
* **handler**: an asynchronous function that must return a truthy value (for example, not `undefined` or `null`)
```

Reply:
As I remember that ‘return null’ also works fine.

Reply:
Duplication happens because keys with undefined values are automatically removed from hydrations payload
```
JSON.stringify({ a: 1, b: undefined })
'{"a":1}'
JSON.stringify({ a: 1, b: null })
'{"a":1,"b":null}'
```

Reply:
https://github.com/nuxt/nuxt/blob/5abd8a231f26a2e08071a59caa87b80eaaa6cd91/packages/nuxt/src/app/composables/asyncData.ts#L145C3-L145C60

Reply:
Keys with undefined values should still be serialised, as we use `devalue`. And the reason I'm suggesting not returning `null` is for safety - see https://github.com/nuxt/nuxt/pull/23351.